packages/graph/hash_graph/doc/structural-queries.md

```diff
@@ -0,0 +1,233 @@
+
+```json5
+{
+  filter: {
+    all: [
+      {
+        notEqual: [{ path: ["description"] }, null],
+      },
+      {
+        not: {
+          equal: [{ path: ["title"] }, { parameter: "Foo" }],
+        },
+      },
+    ],
+  },
+}
+```
+
+An empty `all` filter will always match. An empty `any` filter will never match.
+
+## Depth
+
+The depth determines how many edges away from the root vertices the subgraph will be resolved.
+
+Elements in the subgraph are connected via edges. For example, ontology elements may have references to other records, a property type may reference other property types or data types. The depths provided alongside a filter specify how many steps to explore along a chain of references _of a certain edge kind_. Maning, any chain of property type references will be resolved up to the depth given for property types, and _each_ data type referenced in those property types will in turn start a 'new chain' whose exploration depth is limited by the
+depth given for data types.
+
+A depth of `0` means that no edges are explored for that edge kind.
+
+### Example
+
+- `EntityType1` references \[`EntityType2`, `PropertyType1`]
+- `EntityType2` references \[`PropertyType2`]
+- `PropertyType1` references \[`DataType2`]
+- `PropertyType2` references \[`PropertyType3`, `DataType1`]
+- `PropertyType3` references \[`PropertyType4`, `DataType3`]
+- `PropertyType4` references \[`DataType3`]
+
+if a query on `EntityType1` is made with the following depths:
+
+```json5
+{
+  filter: {
+    // ...
+  },
+  graphResolveDepths: {
+    dataTypeResolveDepth: 1,
+    propertyTypeResolveDepth: 3,
+    entityTypeResolveDepth: 1,
+    entityResolveDepth: 0,
+  },
+}
+```
+
+then the returned subgraph will be:
+
+- `referenced_entity_types`: \[`EntityType2`]
+- `referenced_property_types`: \[`PropertyType1`, `PropertyType2`, `PropertyType3`]
+- `referenced_data_types`: \[`DataType1`, `DataType2`]
+
+Please note, that all depth parameters has to be passed each time a query is made to prevent unexpected results.
+
+# Commonly used queries
+
+This section covers a few commonly used queries. It's leaving out the depth parameter for brevity.
+
+For an exhaustive list of all supported paths, please generate the Rust documentation and refer to the documentation of `DataTypeQueryPath`, `PropertyTypeQueryPath`, `EntityTypeQueryPath`, and `EntityQueryPath`.
+
+## Get a specific type
+
+```json5
+{
+  filter: {
+    all: [
+      {
+        equal: [
+          { path: ["baseUri"] },
+          { parameter: "{{base_uri_of_desired_type}}" },
+        ],
+      },
+      {
+        equal: [
+          { path: ["version"] },
+          { parameter: {{version_of_desired_type}} },
+        ],
+      },
+    ],
+  },
+  graphResolveDepths: {
+    // ...
+  },
+}
+```
+
+## Get all entities owned by a specific user
+
+```json5
+{
+  filter: {
+    equal: [{ path: ["ownedById"] }, { parameter: "{{user_id}}" }],
```

Comment:
Should we use the word account here?
```suggestion
## Get all entities owned by a specific account

```json5
{
  filter: {
    equal: [{ path: ["ownedById"] }, { parameter: "{{account_id}}" }],
```